examples/superdense_coding.py

```diff
@@ -0,0 +1,86 @@
+"""Superdense Coding.
+Superdense Coding is a method to transmit two classical bits of information
+by sending only one qubit of information. This is accomplished by
+pre-sharing an entangled state between the sender and the receiver. This
+entangled state allows the receiver of the one qubit of information to
+decode the two classical bits that were originally encoded by the sender.
+
+In the following example output, a sender sets qubit 0 (i.e., q0) to 0 and
+qubit 1 (i.e., q1) to 1. By sharing one qubit of information (i.e., q2),
+the receiver is able to decode the original 0 and 1 in qubits 3 and 4,
+respectively, when qubits 3 and 4 are measured. This is only possible given
+that an entangled state is pre-shared between the sender and receiver.
+
+In the actual program below, qubit 0 and qubit 1 are randomly set to
+either 1 or 0.
+
+=== REFERENCE ===
+https://en.m.wikipedia.org/wiki/Superdense_coding
+
+=== EXAMPLE OUTPUT ===
+Circuit:
+0: ---------------M-------@-------------------
+                          |
+1: ---X-----------M---@---|-------------------
+                      |   |
+2: -------H---@-------X---@---x---------------
+              |               |
+3: -----------|---------------x---@---H---M---
+              |                   |
+4: -----------X-------------------X-------M---
+
+Results:
+0=0
+1=1
+3=0
+4=1
+
+"""
+
+import random
+import cirq
+
+
+def make_superdense_circuit():
+    circuit = cirq.Circuit()
+    (q0, q1, q2, q3, q4) = cirq.LineQubit.range(5)
+
+    # Randomly sets q0 and q1 to either 0 or 1
+    if random.randint(0, 1):
+        circuit.append([cirq.X(q0)]) # coverage: ignore
+    if random.randint(0, 1):
```

Comment:
How about embedding the randomness into the circuit itself? The gate sequence Hadamard Measure will produce a random bit. We could also use a depolarizing channel to randomize the qubit.

Reply:
Hi @Strilanc, thanks so much for these thoughts.  I really do appreciate you passing them along! I should have some time to incorporate a bit later today. By the way, I am a huge fan of [your website](http://algassert.com). Thanks so much for your work on that as well!

Reply:
You could use one measurement for the inputs `measure(in1, in2, key="input")` and one for the outputs, and then print out the results in a way that made it obvious they agreed.

Reply:
Great suggestion! I just reflected this change in the latest commit, though am still trying to better understand how to parse the TrialResult, so there is probably some further clean-up that can be made down the road.  That said, please do let me know if there are any other suggestions on how to improve this.

Thanks again for all of your help!!!